rdr_service/genomic/genomic_job_components.py

```diff
@@ -646,3 +651,159 @@ def _get_sex_at_birth(self, participant_id):
+        )
+
+    def _get_source_data_query(self, manifest_type):
+        """
+        Returns the query to use for manifest's source data
+        :param manifest_type:
+        :return: query object
+        """
+        query_sql = ""
+        if manifest_type == GenomicManifestTypes.DRC_CVL_WGS:
+            query_sql = """
+                SELECT s.genomic_set_name
+                    , m.biobank_id
+                    , m.sex_at_birth
+                    , m.ny_flag
+                    , gcv.site_id
+                    , NULL as secondary_validation
+                FROM genomic_set_member m
+                    JOIN genomic_set s
+                        ON s.id = m.genomic_set_id
+                    JOIN genomic_gc_validation_metrics gcv
+                        ON gcv.genomic_set_member_id = m.id
+                WHERE gcv.processing_status = "pass"
+                    AND m.reconcile_cvl_job_run_id IS NOT NULL
+                    AND m.cvl_manifest_wgs_job_run_id IS NULL
+            """
+        return query_sql
+
+    def _get_manifest_columns(self, manifest_type):
+        """
+        Defines the columns of each manifest-type
+        :param manifest_type:
+        :return: column tuple
+        """
+        columns = tuple()
+        if manifest_type == GenomicManifestTypes.DRC_CVL_WGS:
+            columns = (
+                "genomic_set_name",
+                "biobank_id",
+                "sex_at_birth",
+                "ny_flag",
+                "site_id",
+                "secondary_validation",
+            )
+        return columns
+
+    def get_def(self, manifest_type):
+        return self.MANIFEST_DEFINITIONS[manifest_type]
+
+
+class ManifestCompiler:
+    """
+    This component compiles Genomic manifests
+    based on definitions provided by ManifestDefinitionProvider
+    TODO: All manifests should be updated to using this pattern, currently only CVL
+    """
+    def __init__(self, run_id, bucket_name=None):
+        self.run_id = run_id
+
+        self.bucket_name = bucket_name
+        self.output_file_name = None
+        self.manifest_def = None
+
+        self.def_provider = ManifestDefinitionProvider(
+            job_run_id=run_id, bucket_name=bucket_name
+        )
+
+        # Dao components
+        self.member_dao = GenomicSetMemberDao()
+        self.metrics_dao = GenomicGCValidationMetricsDao()
+
+    def generate_and_transfer_manifest(self, manifest_type):
+        """
+        Main execution method for ManifestCompiler
+        :return: result code
+        """
+        self.manifest_def = self.def_provider.get_def(manifest_type)
+        source_data = self._pull_source_data()
+        if source_data:
+            self.output_file_name = self.manifest_def.output_filename
+            logging.info(
+                f'Preparing manifest of type {manifest_type}...'
+                f'{self.manifest_def.destination_bucket}/{self.manifest_def.output_filename}'
```

Comment:
We've never defined the logging formatter to use f strings. It will use % behind the scenes. 
See [python formatting](https://docs.python.org/3/library/logging.html#logging.Formatter) for info.  
Until we do, we should use the % style formatting.

Reply:
Google AppEngine 2 logs don't like %-style though, so I updated most of the messages back in https://github.com/all-of-us/raw-data-repository/commit/8ab024f986621eef4dd5a4937505e60dc681c7f4.  I think the default formatter is converted by AppEngine Logging behind the scenes.